datalad_neuroimaging/extractors/fslfeat.py

```diff
@@ -0,0 +1,276 @@
+        '2': 'regulardown',
+        '3': 'sliceordercfgfile',
+        '4': 'slicetimingcfgfile',
+        '5': 'interleaved',
+    },
+    'motionevs': {
+        '0': False,
+        '1': True,
+    },
+    'mixed_yn': {
+        '0': 'me_ols',
+        '1': 'me_flame1+2',
+        '2': 'me_flame1',
+        '3': 'fixedeffects',
+    },
+    'thresh': {
+        '0': 'none',
+        '1': 'uncorrected',
+        '2': 'voxel',
+        '3': 'cluster',
+    },
+    'tempfilt_yn': {
+        '0': False,
+        '1': True,
+    },
+    'shape': {
+        '0': 'square',
+        '1': 'sinusoid',
+        '2': 'custom_ev1',
+        '3': 'custom_ev3',
+        '4': 'interaction',
+        '10': 'empty',
+    },
+}
+
+
+def _convert_dtype(val):
+    if isinstance(val, (bool, int, float)):
+        return val
+    try:
+        return int(val)
+    except:
+        try:
+            return float(val)
+        except:
+            return val
+
+
+def _read_fsf(path, dspath):
+    props = {}
+    evs = {}
+    contrasts = {}
+    disabled = set()
+    valid_props = {}
+    for line in open(path):
+        if not line.startswith('set '):
+            continue
+        cmd, var = line.split()[:2]
+        # TODO match feat_files(...) to get input fMRI data
+        if not (cmd == 'set' and var.startswith('fmri(') and var.endswith(')')):
+            lgr.debug(
+                'Ignoring unknown/malformed setting from %s: %s',
+                path, line)
+            continue
+        if var in _ignor_vars or var.startswith('fmri(conpic_'):
+            continue
+        # cut out value, strip whitespace around, and shed quotes
+        val = line[5 + len(var):].strip().strip('"')
+        # strip common 'fmri(xxx)'
+        var = var[5:-1]
+        store_in = props
+        if var[-1].isdigit() and '.' not in var and not var.startswith('conmask'):
+            ev_id = int(var[-1])
+            # this is a variable describing an EV
+            store_in = evs.get(ev_id, {})
+            evs[ev_id] = store_in
+            var = var[:-1]
+        if val.startswith(dspath):
+            val = relpath(val, start=dspath)
+        # transcode values on a best-effort basis
+        val = _trancode.get(var, {}).get(val, val)
+        # final dtype
+        val = _convert_dtype(val)
+        if var in _procedural_switches:
+            if val == 0:
+                disabled.add(var)
+            continue
+        if var in ('evs_orig', 'evs_real'):
+            valid_props[var] = int(val)
+        if var.endswith('_yn'):
+            var = var[:-3]
+        if val == '':
+            continue
+        # handle specification of vector elements
+        vecvals = _contrastvar_expr.match(var)
+        if vecvals:
+            name, con, pos = vecvals.groups()
+            pos = int(pos)
+            con = int(con)
+            if name == 'ortho':
+                if pos == 0:
+                    # XXX this is a bit strange, there is not EV0
```

Comment:
if that expected not to have it, I would raise ValueError to catch a moment of 'enlightenment' when we run into such case

Reply:
;-) If you point me to the docs on what is to be expected and what not, I'll do that....